### PR TITLE
Experts: Directions / regions filter + page number in URL #239

### PR DESCRIPTION
**Pull request by zolotoho**
develop

## GitHub Board

https://github.com/ita-social-projects/dokazovi-fe/issues/239

[Experts: Directions / regions filter + page number in URL #239 ]( https://github.com/ita-social-projects/dokazovi-fe/issues/239 )

## Summary of issue

- [x]   See changes in PR #216
- [x]   ExpertsView should use useQuery hook to store filters and page number as query params.
- [x]  fetchExperts should use these params.

## Summary of change

- [x]  Changed ExpertView behavior
- [x]  Created new Component
- [x]  Refined ExpertsSlice

### CODE REVIEW

File: src/lib/components/CheckBoxFilterContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+export interface ICheckBoxFilterContainerProps {
+  onFormChange: (checked: ICheckBoxFormState, filterType?) => void;
+  possibleFilters: IFilter[];
+  selectedFilters?: IFilter[];
+  filterTitle: string;
+  filterType?: FilterTypeEnum;
+}
+
+const CheckBoxFilterFormContainer: React.FC<ICheckBoxFilterContainerProps> = ({
```
Comment (SergeOlabin) [inline src/lib/components/CheckBoxFilterContainer.tsx:32]:
```suggestion
const CheckBoxDropdownFilterForm: React.FC<ICheckBoxFilterContainerProps> = ({
```
name is better

File: src/modules/experts/components/ExpertsView.tsx [hunk 2]
```diff
@@ -35,44 +50,104 @@ const ExpertsView: React.FC = () => {
...
+    const DirectionsQuery = query.get(DIRECTIONS_QUERY);
+
+    const isPage = pageQuery ? pageQuery - 1 : 0;
+    dispatch(setExpertsPage(isPage));
+
+    const regionsType = {};
+    regionsQuery?.split(',').forEach((el) => {
+      regionsType[el] = true;
+    });
+    dispatch(setExpertsRegionsFilter({ value: regionsType }));
```
Comment (SergeOlabin) [inline src/modules/experts/components/ExpertsView.tsx:65]: don't set the value to the store

File: src/modules/experts/components/ExpertsView.tsx [hunk 3]
```diff
@@ -35,44 +50,104 @@ const ExpertsView: React.FC = () => {
...
-  const handlePageChange = (_, page: number) => {
-    if (page === 1) {
-      dispatch(setExpertsPage(0));
+    query.set(queryType, checkedIds.join(','));
+    if (!checkedIds.length) {
+      query.delete(queryType);
     } else {
-      dispatch(setExpertsPage(page - 1));
+      query.set(queryType, checkedIds.join(','));
     }
```
Comment (SergeOlabin) [inline src/modules/experts/components/ExpertsView.tsx:88]: else statement seems obsolete, it's already on line : 83

File: src/modules/experts/components/ExpertsView.tsx [hunk 4]
```diff
@@ -35,44 +50,104 @@ const ExpertsView: React.FC = () => {
...
+    regionsQuery?.split(',').forEach((el) => {
+      regionsType[el] = true;
+    });
+    dispatch(setExpertsRegionsFilter({ value: regionsType }));
+
+    const directionsType = {};
+    DirectionsQuery?.split(',').forEach((el) => {
+      directionsType[el] = true;
+    });
+    dispatch(setExpertsDirectionsFilter({ value: directionsType }));
```
Comment (SergeOlabin) [inline src/modules/experts/components/ExpertsView.tsx:71]: don't set the value to the store

File: src/modules/experts/components/ExpertsView.tsx [hunk 5]
```diff
@@ -35,44 +50,104 @@ const ExpertsView: React.FC = () => {
...
+    const regionsQuery = query.get(REGIONS_QUERY);
+    const DirectionsQuery = query.get(DIRECTIONS_QUERY);
+
+    const isPage = pageQuery ? pageQuery - 1 : 0;
+    dispatch(setExpertsPage(isPage));
+
+    const regionsType = {};
+    regionsQuery?.split(',').forEach((el) => {
+      regionsType[el] = true;
+    });
```
Comment (SergeOlabin) [inline src/modules/experts/components/ExpertsView.tsx:64]: use reduce

File: src/modules/experts/components/ExpertsView.tsx [hunk 6]
```diff
@@ -35,44 +50,104 @@ const ExpertsView: React.FC = () => {
...
+    dispatch(setExpertsRegionsFilter({ value: regionsType }));
+
+    const directionsType = {};
+    DirectionsQuery?.split(',').forEach((el) => {
+      directionsType[el] = true;
+    });
```
Comment (SergeOlabin) [inline src/modules/experts/components/ExpertsView.tsx:70]: use reduce

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/lib/components/CheckBoxFilterContainer.tsx (hunk 1 above)
Comment (zolotoho) [reply src/lib/components/CheckBoxFilterContainer.tsx:32]: Done

File: src/modules/experts/components/ExpertsView.tsx (hunk 2 above)
Comment (zolotoho) [reply src/modules/experts/components/ExpertsView.tsx:65]: Done, but without pageNumber

File: src/modules/experts/components/ExpertsView.tsx (hunk 3 above)
Comment (zolotoho) [reply src/modules/experts/components/ExpertsView.tsx:88]: Fixed and improved

File: src/modules/experts/components/ExpertsView.tsx (hunk 4 above)
Comment (zolotoho) [reply src/modules/experts/components/ExpertsView.tsx:71]: Done

File: src/modules/experts/components/ExpertsView.tsx (hunk 5 above)
Comment (zolotoho) [reply src/modules/experts/components/ExpertsView.tsx:64]: Did without reduce and without forEach

File: src/modules/experts/components/ExpertsView.tsx (hunk 6 above)
Comment (zolotoho) [reply src/modules/experts/components/ExpertsView.tsx:70]: Done

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**